In [1]:
import numpy as np
import pandas as pd
from functions import *

In [20]:
df = pd.read_csv("data/gielda.csv", sep=";")
df["IndexReturnPerc"] = df["IndexReturnPerc"].apply(
    lambda x: float(x.replace(",", "."))
)
df["StockReturnPerc"] = df["StockReturnPerc"].apply(
    lambda x: float(x.replace(",", "."))
)

df.head()

Date  IndexReturnPerc  StockReturnPerc
0  2020-12-31            10.36             8.41
1  2021-01-31            -3.67             0.53
2  2021-02-28             2.53             2.63
3  2021-03-31            -1.56            15.00
4  2021-04-30            -1.37            -2.90

In [22]:
# correlation between the height and head size in the sample:
from scipy import stats

correlation_test = stats.pearsonr(df["IndexReturnPerc"], df["StockReturnPerc"])
print(correlation_test)

PearsonRResult(statistic=np.float64(0.34894141962667125), pvalue=np.float64(0.05876971566370387))


as alpha =0.1, we can reject the null hypothesis. Thre is a linear correlation between index and stock returns.

In [23]:
confidence_interval = correlation_test.confidence_interval(0.90)

print(
    "Test:\n\n", correlation_test, "\n\n Confidence interval: \n\n", confidence_interval
)

Test:

 PearsonRResult(statistic=np.float64(0.34894141962667125), pvalue=np.float64(0.05876971566370387)) 

 Confidence interval: 

 ConfidenceInterval(low=np.float64(0.04764956364660326), high=np.float64(0.5920328236623378))


we are 90% confident that correlation between index and stock returns is between 4.76% and 59.2%.